mce-hal.c

```diff
@@ -38,7 +38,7 @@ product_id_t get_product_id(void)
 	gchar *tmp = NULL;
 
 	if (product_id != PRODUCT_UNSET)
-		goto EXIT;
```

Comment:
Does this fix any compiler warnings or errors? Or did you want to get rid of the goto?

Reply:
Excessive goto's in high level programming languages are considered harmful.
It's better to avoid using goto without a reason.

It's not a taboo. I'm using goto's when it makes sense in the code to make it more simple and/or easier to read. But this is not the case.

https://github.com/maemo-leste/status-area-applet-battery/commit/761aef615286241a32c07a765e221e63dcc0fdc3
http://www.cs.utexas.edu/users/EWD/ewd02xx/EWD215.PDF
https://stackoverflow.com/questions/3517726/what-is-wrong-with-using-goto

Reply:
Sure, but I am not sure if it made sense to remove them here. I don't care much either way, but when I modify functions to add functionality, I try to stick to how the function originally worked, unless it's clearly broken. :p 

In this case, nothing was added to this function, as far as I can tell. You just moved some comments and goto into return - as opposed to the other commits that fix compiler warnings. :)

Reply:
That's why I called the commit "Minor fixes": https://github.com/maemo-leste/mce/pull/3/commits/065823f5d8773ea6ae080c0d5958d356dad73855

Sorry, if I didn't get the idea from your last message correct. Could you please explain?

Reply:
On the face, nothing was particularly wrong with this function. It had no logic mistakes, and the comment was maybe on a line that was too long, but it was in context. I would have probably just left it as is, and let `indent` at some point take care of it.

But it's fine this way, if you move the comment more in context.

Reply:
Sorry, the comment part was part of the change above, not here.

Reply:
> Sorry, the comment part was part of the change above, not here.

AFAIK, you can just remove the wrong comment from github.

Reply:
test